atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/PersistentLockManagerTest.java

```diff
@@ -119,4 +124,19 @@ public void shutdownWithoutAcquireIsNoOp() {
 
         verifyZeroInteractions(mockPls);
     }
+
+    @Test
+    public void doesNotDeadlockOnShutdownIfLockCannotBeAcquired() throws InterruptedException {
```

Comment:
Should have a timeout on this test, to avoid it hanging if we break the "don't deadlock" thing again.